pkg/webhooks/pod_admission_webhook.go

```diff
@@ -0,0 +1,64 @@
+package webhooks
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+
+	jobset "sigs.k8s.io/jobset/api/jobset/v1alpha2"
+	"sigs.k8s.io/jobset/pkg/controllers"
+	"sigs.k8s.io/jobset/pkg/util/shared"
+)
+
+//+kubebuilder:webhook:path=/validate--v1-pod,mutating=false,failurePolicy=fail,sideEffects=None,groups="",resources=pods,verbs=create,versions=v1,name=vpod.kb.io,sideEffects=None,admissionReviewVersions=v1
+
+func (p *podMutatingWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
```

Comment:
Handle is a generic handler for both mutating and admission calls. We already overridden the mutating handler using by implementing the Default interface. We should do the same for Validate. Note that the validating callback is made on create, update and delete, we only need to validate on create.

Here is an example: https://github.com/kubernetes-sigs/controller-runtime/blob/main/examples/builtins/validatingwebhook.go

the ValidateUpdate and ValidateDelete will simply return nil, nil in our case.

Reply:
I see, thanks! Done